value-processor/src/org/immutables/value/processor/meta/Constitution.java

```diff
@@ -62,7 +63,14 @@ public Generics generics() {
 
 	@Value.Derived
 	public Visibility implementationVisibility() {
-		if (style().visibility() == ImplementationVisibility.PRIVATE
+		ImplementationVisibility visibility;
+		if (!Strings.isNullOrEmpty(style().visibilityString())) {
```

Comment:
never should be null, better use `.isEmpty()`.  In general, usage of `isNullOrEmpty` and similar (on collections, strings) considered a code smell (even its javadoc suggest alternatives)

Reply:
makes sense, updated